lib/src/ovis_event/ovis_event.c

```diff
@@ -334,6 +334,26 @@ void ovis_scheduler_free(ovis_scheduler_t m)
 	ovis_scheduler_ref_put(m);
 }
 
+static void set_next_periodic_time(ovis_event_t ev, const struct timeval *tv)
+{
+	uint64_t now, us;
+	int64_t diff;
+	now = us = tv->tv_sec * USEC + tv->tv_usec;
+	us = ROUND(us, ev->param.periodic.period_us);
+	us += ev->param.periodic.phase_us;
+	diff = (int64_t)us - (int64_t)now;
+	if (diff <= 0) {
+		/* This is possible if phase is negative */
+		diff += ev->param.periodic.period_us;
+	} else if (diff > ev->param.periodic.period_us) {
+		/* Prevent waiting > 1 period */
```

Comment:
detecting a wait of > 1 period should probably generate a warning. whether that warning is issued here or higher up where the user sets the improper period in the first place is a good question. probably the latter would be less noisy.